src/mscorlib/src/System/GC.cs

```diff
@@ -292,6 +292,17 @@ internal static int CollectionCount (int generation, bool getSpecialGCCount)
         // In this code, Foo will be finalized in the middle of 
         // stream.MethodThatSpansGCs, thus closing a stream still in use."
         //
+        // The problem persists in "release" configuration when code runs without
+        // debugger even if the user explicitly binds a reference to the object:
```

Comment:
The comment about debugger is not correct - you can definitely see this problem in even in release configuration when running with debugger attached.